src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -292,8 +296,21 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 			ereport(ERROR, (errmsg("grouped column list cannot be hashed")));
 		}
 
-		/* switch to hashed aggregate strategy to allow grouping */
-		aggregateStrategy = AGG_HASHED;
+		if (hasDistinctAggregate)
+		{
+			/*
```

Comment:
generally, we put comments outside of the if instead of inside

Reply:
ok